### PR TITLE
Add sys_fallocate

### PR DESCRIPTION
**Pull request by lucassong-mh**

### CODE REVIEW

File: kernel/aster-nix/src/fs/inode_handle/mod.rs [hunk 1]
```diff
@@ -184,6 +185,36 @@ impl InodeHandle_ {
         self.dentry.inode().poll(mask, poller)
     }
 
+    fn fallocate(&self, flags: FallocateFlags, offset: usize, len: usize) -> Result<()> {
+        let access_mode = self.access_mode();
+        if !access_mode.is_writable() {
+            return_errno_with_message!(Errno::EBADF, "try to fallocate a read-only file");
+        }
```
Comment (liqinggd) [inline kernel/aster-nix/src/fs/inode_handle/mod.rs:192]: Please move this `access_mode` checking code to `dyn_cap.rs`.

Comment (lucassong-mh) [reply kernel/aster-nix/src/fs/inode_handle/mod.rs]: Make sense and done

**Comment by liqinggd** [comment]
LGTM

**Review by tatetian — COMMENTED** [review]
Just one tiny issue before merging this PR.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**